vcd/resource_vcd_rde_type.go

```diff
@@ -0,0 +1,337 @@
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				Description: "To be used when creating a new version of a Runtime Defined Entity Type. Specifies the version of the type that will be the template for the authorization configuration of the new version." +
+					"The Type ACLs and the access requirements of the Type Behaviors of the new version will be copied from those of the inherited version." +
+					"If not set, then the new type version will not inherit another version and will have the default authorization settings, just like the first version of a new type",
+			},
+			"readonly": {
+				Type:        schema.TypeBool,
+				Computed:    true,
+				Description: "True if the Runtime Defined Entity Type cannot be modified",
+			},
+		},
+	}
+}
+
+// hasJsonValueChanged tells Terraform whether the JSON schema set in HCL configuration (which can have whatever identation and other quirks)
+// matches the obtained JSON from VCD.
+func hasJsonValueChanged(key, oldValue, newValue string, _ *schema.ResourceData) bool {
+	areEqual, err := areMarshaledJsonEqual([]byte(oldValue), []byte(newValue))
+	if err != nil {
+		util.Logger.Printf("[ERROR] Could not compare JSONs for computing difference of %s: %s", key, err)
+		return false
+	}
+	return areEqual
+}
+
+func resourceVcdRdeTypeCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	vcdClient := meta.(*VCDClient)
+
+	jsonSchema, err := getRdeTypeSchema(d)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	executeRdeTypeFunctionWithMutex(func() {
+		_, err = vcdClient.VCDClient.CreateRdeType(&types.DefinedEntityType{
+			Name:             d.Get("name").(string),
+			Nss:              d.Get("nss").(string),
+			Version:          d.Get("version").(string),
+			Description:      d.Get("description").(string),
+			ExternalId:       d.Get("external_id").(string),
+			InheritedVersion: d.Get("inherited_version").(string),
+			Interfaces:       convertSchemaSetToSliceOfStrings(d.Get("interface_ids").(*schema.Set)),
+			IsReadOnly:       d.Get("readonly").(bool),
+			Schema:           jsonSchema,
+			Vendor:           d.Get("vendor").(string),
+		})
+	})
+
+	if err != nil {
+		return diag.Errorf("could not create the Runtime Defined Entity Type: %s", err)
+	}
+	return resourceVcdRdeTypeRead(ctx, d, meta)
+}
+
+// getRdeTypeSchema gets the schema as string from the Terraform configuration
+func getRdeTypeSchema(d *schema.ResourceData) (map[string]interface{}, error) {
+	var jsonSchema string
+	var err error
+	if url, isUrlSet := d.GetOk("schema_url"); isUrlSet {
+		jsonSchema, err = fileFromUrlToString(url.(string), ".json")
+		if err != nil {
+			return nil, fmt.Errorf("could not download JSON schema from url %s: %s", url, err)
+		}
+	} else {
+		jsonSchema = d.Get("schema").(string)
+	}
+
+	var unmarshalledJson map[string]interface{}
+	err = json.Unmarshal([]byte(jsonSchema), &unmarshalledJson)
+	if err != nil {
+		return nil, err
```

Comment:
the error message could give some context ("failed unmarshalling schema: %s" or something similar) here as it will be returned bare without any context if it fails here

Reply:
Fixed, thanks!